miqi/runtime/task_runner.py

```
@@ -697,6 +697,35 @@ async def _handle_user_message(self, msg: UserMessage) -> None:
             )
 
+        # ── #740: resume context — continue an interrupted turn ──────
+        # Replan (not replay): feed the snapshot's half-generated content to
+        # the model as context and instruct it to continue from where it
+        # stopped, rather than restoring raw messages.
+        resume_turn_id = getattr(msg, "resume_turn_id", None)
+        resume_snapshot: dict[str, Any] | None = None
+        if resume_turn_id and history_runtime is not None:
+            try:
+                resume_snapshot = await history_runtime.get_snapshot(resume_turn_id)
+            except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)